sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeFormatterHelper.scala

```diff
@@ -31,17 +31,39 @@ import org.apache.spark.sql.internal.SQLConf
 import org.apache.spark.sql.internal.SQLConf.LegacyBehaviorPolicy._
 
 trait DateTimeFormatterHelper {
+  private def getOrDefault(accessor: TemporalAccessor, field: ChronoField, default: Int): Int = {
+    if (accessor.isSupported(field)) {
+      accessor.get(field)
+    } else {
+      default
+    }
+  }
+
+  protected def toLocalDate(temporalAccessor: TemporalAccessor): LocalDate = {
+    val year = getOrDefault(temporalAccessor, ChronoField.YEAR, 1970)
```

Comment:
How often do users consider 1970 as the default year, your guess? I would propose current year as we do in stringToTimestamp:
https://github.com/apache/spark/blob/ebc8fa50d0422f3db47b2c45025c7f2efe6ee39a/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala#L396

Reply:
It's from 2.4. I don't care too much about it as long as we have a default value,

Reply:
It is some kind of inconsistent behaviour between CAST and parsing via patterns (in to_timestamp for example). Should we document somewhere the default year?